File: 4.Clients/WebApi/ClientApp/src/entities/days-off.ts
import { Employee } from "./employee";
import { DaysOffTypeEnum } from './enums/daysoff-type.enum';
import { DaysOffStatusEnum } from './enums/daysoff-status.enum';

export class DaysOff {
    id: number;
    date: Date; // RANGE PICKER
    endDate: Date; // RANGE PICKER
    type: DaysOffTypeEnum; // studyDays, Holydays, Training, PTO
    status: DaysOffStatusEnum; // in review, accepted
    employeeId: number;
    employee: Employee;
}

File: 1.Domain/Domain.Model/DaysOff.cs
using Core;
using Domain.Model.Enum;
using System;
using System.Collections.Generic;

namespace Domain.Model
{
    public class DaysOff : Entity<int>
    {

        public DaysOffStatus Status { get; set; }

        public DateTime Date { get; set; }
        public DateTime EndDate { get; set; }

        public DaysOffType Type { get; set; }

        public int EmployeeId { get; set; }

        public Employee Employee { get; set; }

        public string GoogleCalendarEventId { get; set; }

    }
}
